[2025-07-21T08:01:24.638+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-21T08:01:24.665+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: fetch_weather_foreacst_dag.fetch_forecast manual__2025-07-21T08:01:21.757560+00:00 [queued]>
[2025-07-21T08:01:24.673+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: fetch_weather_foreacst_dag.fetch_forecast manual__2025-07-21T08:01:21.757560+00:00 [queued]>
[2025-07-21T08:01:24.674+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 3
[2025-07-21T08:01:24.686+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): fetch_forecast> on 2025-07-21 08:01:21.757560+00:00
[2025-07-21T08:01:24.693+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'fetch_weather_foreacst_dag', 'fetch_forecast', 'manual__2025-07-21T08:01:21.757560+00:00', '--job-id', '81', '--raw', '--subdir', 'DAGS_FOLDER/weather_dag.py', '--cfg-path', '/tmp/tmprg8iumle']
[2025-07-21T08:01:24.696+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=190) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-21T08:01:24.697+0000] {standard_task_runner.py:91} INFO - Job 81: Subtask fetch_forecast
[2025-07-21T08:01:24.697+0000] {standard_task_runner.py:63} INFO - Started process 192 to run task
[2025-07-21T08:01:24.744+0000] {task_command.py:426} INFO - Running <TaskInstance: fetch_weather_foreacst_dag.fetch_forecast manual__2025-07-21T08:01:21.757560+00:00 [running]> on host f5776ee018f1
[2025-07-21T08:01:24.833+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='fetch_weather_foreacst_dag' AIRFLOW_CTX_TASK_ID='fetch_forecast' AIRFLOW_CTX_EXECUTION_DATE='2025-07-21T08:01:21.757560+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-21T08:01:21.757560+00:00'
[2025-07-21T08:01:24.834+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-21T08:01:24.847+0000] {weather_dag.py:50} INFO - Bắt đầu gọi API thời tiết...
[2025-07-21T08:01:35.582+0000] {weather_dag.py:95} INFO - Các cột có giá trị None: name                           0
forecast_time                  0
temperature                    0
temperatureApparent            0
dewPoint                       0
humidity                       0
windSpeed                      0
windDirection                  0
windGust                       0
pressureSurfaceLevel           0
cloudCover                     0
precipitationProbability       0
uvIndex                      190
ezHeatStressIndex           1200
dtype: int64
[2025-07-21T08:01:35.585+0000] {weather_dag.py:99} WARNING - Cột name vẫn chứa chuỗi 'nan'!
[2025-07-21T08:01:35.599+0000] {weather_dag.py:99} WARNING - Cột uvIndex vẫn chứa chuỗi 'nan'!
[2025-07-21T08:01:35.628+0000] {weather_dag.py:111} INFO - name                        object
forecast_time               object
temperature                 object
temperatureApparent         object
dewPoint                    object
humidity                     int64
windSpeed                   object
windDirection                int64
windGust                    object
pressureSurfaceLevel        object
cloudCover                   int64
precipitationProbability     int64
uvIndex                     object
ezHeatStressIndex           object
dtype: object
[2025-07-21T08:01:36.182+0000] {weather_dag.py:134} INFO - Đã insert 1200 dòng dữ liệu thời tiết
[2025-07-21T08:01:36.186+0000] {python.py:237} INFO - Done. Returned value was: None
[2025-07-21T08:01:36.186+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-21T08:01:36.195+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=fetch_weather_foreacst_dag, task_id=fetch_forecast, run_id=manual__2025-07-21T08:01:21.757560+00:00, execution_date=20250721T080121, start_date=20250721T080124, end_date=20250721T080136
[2025-07-21T08:01:36.225+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-07-21T08:01:36.238+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-21T08:01:36.240+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
